### ISSUE TITLE
Hotkey to force restore

### ISSUE THREAD

**Issue by okgolove** [relikd/Memmon#5]
Hello! Thank you for the great app!
It would be very helpful if I was able to use some keys combination to trigger restoring. That would partially resolve issue related to spaces.

**Comment by relikd** [comment]
Hi, lets think this through first. Can you list the steps how you would like to restore windows? Currently all window positions are saved whenever a monitor is plugged in/out. After that, restore the windows from a previous session with less/more monitors. I am not sure if it would help to manually trigger a restore-windows-function. 'Cause at this point either the wrong positions are already saved, or the restore did actually restore the windows. Do you want a “save” as well as “restore” hotkey? – or to put it differently: how do you think a manual restore will help you? vs. automatically triggering the same action as soon as the monitor count changes?
The windows in a space should be restored as soon as the space becomes active.

**Comment by okgolove** [comment]
[DisplayMaid](https://apps.apple.com/us/app/display-maid/id450063525?mt=12) works the next way: you save your windows position (for every space) and then you can restore the state using hotkey combination on every space. That was the main idea.

I'm still not able to understand how memmon works with spaces. I'm on space 2, disconnect ex. display, switch to space 3, then connect display, it restores nothing. As soon as I switch to space 1 it restores windows properly. Switch to space 2 or 3 again - nothing happens. Should it restore windows automatically for every space as soon as I switch to them?
If this part worked as expected I would not think about manually functionality.

**Comment by relikd** [comment]
Yes, it should work exactly that way. Memmon should save the stat of all windows of all spaces as soon as you unplug or attach a new monitor. But it will only restore the windows of a space as soon as you navigate to that space (limitations of AXUIElement). This is also what DisplayMaid says:

"Display Maid can’t save or restore positions for windows that are not in the currently active space. This is because there isn't currently a way provided to interact with windows that are in hidden spaces. "

But, Memmon should indeed restore the windows once you visit a space. Can you check whether all windows are "forgotten" or if it happens only to specific apps? I noticed that Firefox /  Chrome may have issues with restoring the windows. But Finder windows should work fine.

**Comment by okgolove** [comment]
This is about all windows. Btw, do you use separate spaces per display?

**Comment by relikd** [comment]
As of now I have no access to an external monitor. So I can't really debug what is going on. What do you mean with seperate spaces per display? If you use spaces, each display will have the same number of spaces. Or in other words, each space has as many display as are connected. During my testing a few months ago I had a setup with one external display and my internal Macbook display. I would place my windows, with the display attached, in two spaces. Then I detach the external display and all the windows get moved to the internal display. I also tested what happens if I switch between spaces at this stage and after the next stage. Next, I reattach the display and the windows of the current space get re-set again to their original position. If I switch to another space, the windows do get re-set there too. However it took one or two seconds before macOS got the notification that a space-change was performed (always a little shock fearing it wouldnt fix it).

**Comment by okgolove** [comment]
I'm talking about this setting
![image](https://user-images.githubusercontent.com/36846182/154090784-4efbb673-3d4c-4167-b93d-469f4b196798.png)


Will try to test this code soon, thank you for responsiveness!

**Comment by okgolove** [comment]
I've tried to run modified app but it stopped work at all.

**Comment by relikd** [comment]
ah, yeah, you'll need to add Terminal (or whatever app you are using to run the script) to "System Preference" > "Security & Privacy" > "Accessibility".

**Comment by okgolove** [comment]
I've already done it. Still no luck.
